atomate/vasp/firetasks/exchange.py

```diff
@@ -0,0 +1,352 @@
+
+        # Get MSONable Heisenberg Model
+        hmodel = hmapper.get_heisenberg_model()
+        hmodel_dict = hmodel.as_dict()
+
+        # Update FW spec with model
+        name = "heisenberg_model_" + str(self["cutoff"])
+        update_spec = {name: hmodel}
+
+        # Write to file
+        dumpfn(hmodel_dict, name + ".json")
+
+        return FWAction(update_spec=update_spec)
+
+
+@explicit_serialize
+class HeisenbergModelToDb(FiretaskBase):
+    """
+    Insert Heisenberg Model object into a DB. Assumes you are in a
+    directory with a model written to a .json file.
+
+    Args:
+        db_file (str): path to file containing the database credentials.
+        wf_uuid (int): Unique id for record keeping.
+        parent_structure (Structure): magnetic structure.
+        cutoff (float): NN search cutoff (Angstrom).
+        tol (float): distance tolerance for similar bonds.
+        average (bool): <J> only or full NN, NNN HeisenbergModel.
+
+    """
+
+    required_params = [
+        "db_file",
+        "wf_uuid",
+        "parent_structure",
+        "cutoff",
+        "tol",
+        "average",
+    ]
+
+    def run_task(self, fw_spec):
+
+        db_file = env_chk(self["db_file"], fw_spec)
+        wf_uuid = self["wf_uuid"]
+        formula = self["parent_structure"].formula
+        formula_pretty = self["parent_structure"].composition.reduced_formula
+
+        name = "heisenberg_model_" + str(self["cutoff"])
+        hmodel = loadfn(name + ".json")
+        #hmodel = HeisenbergModel.from_dict(hmodel_dict)
+
+        # Exchange collection
+        mmdb = VaspCalcDb.from_db_file(db_file, admin=True)
+        mmdb.collection = mmdb.db["exchange"]
+
+        hmodel_dict = hmodel.as_dict()
+        name = "heisenberg_model_" + str(self["cutoff"])
+
+        wf_meta = {"wf_uuid": wf_uuid}
+        task_doc = {
+            "wf_meta": wf_meta,
+            "formula_pretty": formula_pretty,
+            "nn_cutoff": self["cutoff"],
+            "nn_tol": self["tol"],
+            "heisenberg_model": hmodel_dict,
+        }
+
+        if fw_spec.get("tags", None):
+            task_doc["tags"] = fw_spec["tags"]
+
+        mmdb.collection.insert_one(task_doc)
+
+
+@explicit_serialize
+class HeisenbergConvergence(FiretaskBase):
+    """
+    Quick check to see if the Heisenberg model has "converged" for any
+    particular nearest neighbor cutoff value in the sense that |J_ij| < |E_0|
+    for all i, j. 
+
+    If not, it doesn't make sense to do Monte Carlo and the system is either
+    1) not well described by the Heisenberg Model (perhaps the magnetic
+    moments are not localized) or 2) not correctly captured by the provided
+    low-energy magnetic orderings.
+
+    Args:
+        db_file (str): path to file containing the database credentials.
+        wf_uuid (int): Unique id for record keeping.
+        parent_structure (Structure): Structure object.
```

Comment:
Can you double check whether parent_structure is actually required. Again, it seems like it is only used for formula_pretty which itself isn't used anywhere else.

If you do need the parent structure, can you get it from the database without requiring it as a parameter.

Reply:
Yep same thing, the model is being grabbed from the db and parent structure isn't needed. I'll take that out where it isn't needed.